### ISSUE TITLE
Philips Smart LED Bulb not possible to control

### ISSUE THREAD

**Issue by semaf** [YinHangCode/homebridge-mi-philips-light#10]
I have problem with the Bulb and can not control it.

```
[2017-11-26 03:45:13] [MiPhilipsLightPlatform] [MiPhilipsLightPlatform][ERROR]MiPhilipsSmartBulb - Light - getPower Error: Error: Call to device timed out
[2017-11-26 03:45:13] [MiPhilipsLightPlatform] [MiPhilipsLightPlatform][ERROR]MiPhilipsSmartBulb - Light - getBrightness Error: Error: Call to device timed out
[2017-11-26 03:45:13] [MiPhilipsLightPlatform] [MiPhilipsLightPlatform][ERROR]MiPhilipsSmartBulb - Light - getColorTemperature Error: Error: Call to device timed out
```

I read my token from my iPhone and it is 96byte long. Set all of it but got still the error. Shorted down to 32byte and still the same Error.

**ZEXTFWVERSION: 1.3.0_0033
ZMODEL: philips.light.bulb
ZTOKENENCRYPT: 1
ZPERMITLEVEL: 16**

```
{
	"platform": "MiPhilipsLightPlatform",
	"deviceCfgs": [
		{
		    "type": "MiPhilipsSmartBulb",
		    "ip": "192.168.0.31",
		    "token": "96 byte token",
		    "lightName": "living room bulb",
		    "lightDisable": false
		}
	]
}
```

I resetted the Bulb and connected again and read out the Token and it was the same.

**Comment by Kuglikrug** [comment]
Hey @semaf 

I have the exact same problem. Converted the 96 character token to a 32 character one with this website (http://aes.online-domain-tools.com) and used the 32 character one in my config.

The lamp shows up in HomeKit for a couple seconds but then disappears and i get the same error.

**Comment by semaf** [comment]
How did you convert it? I am missing more info to convert it https://aesencryption.net/

**Comment by Kuglikrug** [comment]
here are my settings:
<img width="745" alt="bildschirmfoto 2017-11-26 um 12 08 03" src="https://user-images.githubusercontent.com/34000515/33239469-d2d12bcc-d2a2-11e7-9107-21b685950ab2.png">

the bottom right numbers are your result

**Comment by Kuglikrug** [comment]
can everybody now control my bulb because i uploaded my token btw? 🤔

**Comment by semaf** [comment]
I dont think so because your MAC is important but remove the line.

**Comment by semaf** [comment]
Ok my 32byte was wrong and your way has also same result :( Maybe the firmware of the bulb is the reason. What is the firmware version of your bulb?

**Comment by semaf** [comment]
its working nowww... yippiiiiiiii

**Comment by Kuglikrug** [comment]
hooooooooooooooow? hahahaha

**Comment by semaf** [comment]
Will explain now step by step

**Comment by Kuglikrug** [comment]
i'm hyped 😅 !!!!

**Comment by semaf** [comment]
Check post above

**Comment by Kuglikrug** [comment]
which version of node are you running now? I'm on a Raspberry Pi Zero W and have to update nodejs manually. I am currently on 8.9.0

**Comment by semaf** [comment]
8.9.1

**Comment by semaf** [comment]
If you can remove Node than install it with below and you get latest:

`curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -`

**Comment by Kuglikrug** [comment]
the script does not work with the Zero W because its on arm6 and not arm7. I will manually update to 8.9.1 and let you know if it works

**Comment by semaf** [comment]
Control by Home Assistant is also working fine.

**Comment by Kuglikrug** [comment]
Ok first things first, it works!!

After switching Node Versions back and fourth (apparently there ist no 8.9.1. for Arm6l Chips only 8.9.0 or 9.0.0 and above) i messed up homebridge and nodejs and npm and nothing worked.

Now i have node v 8.7.0 (randomly selected one) and somehow miio --discover now worked.
So I used it to find my token just to make sure i had the right one aaaaaaaaaaaand my original token (the one via iOS Backup) was wrong. 
So I used the correct token and sticked to 8.7.0 and voila, no crash.

Thank you @semaf and thank you @YinHangCode you both did a fantastic job and i am now really really enjoying my Philips LED 😄 

I am so happy now I cant stop smiling when I change the brightness with Siri 🎉 😃 

Greetings from Berlin, Germany
